### DISCUSSION TITLE
add support for mmCIF format

### DISCUSSION THREAD

**Discussion by franke-biosaxs** [AUSAXS/AUSAXS#109] [Ideas]
```
% bin/saxs_fitter 6lyz.cif lyzexp.dat 
Running AUSAXS v1.0.3
Unknown PDB extensions: ./6lyz.cif and ./lyzexp.dat
libc++abi: terminating due to uncaught exception of type std::exception: std::exception
zsh: abort      bin/saxs_fitter 6lyz.cif lyzexp.dat
```
mmCIF is the de-facto standard of the PDB since 2019. A new application should support this file type.

**Comment by klytje** [comment]
I agree. I am but a single developer, so my time has been stretched thin developing this project from scratch alone. I will prioritize adding this in a future (hopefully not too distant) update.

**Comment by klytje** [comment]
A proper mmcif reader will also allow better parsing of the downloaded residue specification files. Currently, structures containing the residue `HEC` crashes since its specification contains quotation marks in the line:
`HEC NA   "N A" N  0 1 Y N N N N N 16.951 28.824 76.033 ? ? ? NA   HEC 6  `

Though the current approach can be extended to support this, it seems to me like adding proper mmcif support is the way to go.

**Comment by klytje** [comment]
mmCIF format is now supported as of commit c6b914d73ae844d8e5e25d91035e581f1e39e562, though it will likely need additional user-testing.
